src/lib/storage/segment_accessor.hpp

```diff
@@ -37,6 +37,8 @@ std::unique_ptr<AbstractSegmentAccessor<T>> create_segment_accessor(const std::s
  *
  *   const std::optional<T> get_typed_value(const ChunkOffset chunk_offset) const;
  *
+ * Accessors are not guaranteed to be thread-safe. For multiple threads that access the same segment, create one
+ * accessor each.
```

Comment:
1) https://www.grammar.cl/english/all-every-each.htm

> Object + each
> 
> Instead of going before/after the subject, each can appear after the object.
> This is common when the noun object refers to an amount, how many of something there is or when giving a price.
> 
>     My paintings are worth $100 each (= Each of my paintings are worth $100)
>     I gave my children each an ice cream.

2) That is exactly the point of this PR. Multiple threads need separate accessors.